navigator_engine/api.py

```diff
@@ -0,0 +1,75 @@
+from flask import Blueprint, jsonify, request
+
+api_blueprint = Blueprint('main', __name__, url_prefix='/api/')
+
+
+@api_blueprint.route('/decision-engine/create', methods=['POST'])
+def decision_engine_create():
+    return jsonify({"message": "success"})
+
+
+@api_blueprint.route('/decision-engine/<engine_id>/progress')
+def decision_engine_progress(engine_id):
+    return jsonify({
+        "milestones":
```

Comment:
We need to add here an overall percentage progress.  
Also, we have a means of determining whether the list of milestone is a complete list, or whether some are missing from the list due to fork logic making it unclear which milestones will need to be completed.  Might suggest we update our API definition to be something like:
```suggestion
    return jsonify({
        "progress": 40,
        "milestone_list_is_complete": True,
        "milestones":
```